custom_components/adaptive_lighting/switch.py

```diff
@@ -557,7 +557,7 @@ def _expand_light_groups(hass: HomeAssistant, lights: list[str]) -> list[str]:
 
 def _supported_features(hass: HomeAssistant, light: str):
     state = hass.states.get(light)
-    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    supported_features = state.attributes.[ATTR_SUPPORTED_FEATURES]
```

Comment:
This is invalid Python syntax now. The problem is probably that `light` does not exist in `hass.states`, so because you use `.get` in line 559, `state = None`.

Reply:
Thank you. I was getting very distracted by the many new error messages I wasn't used to seeing.

Reply:
I'm stumped on this one. I'm currently just googling around how async works. Compared this pr's switch.py to master and went through every change several times now.

It definitely works on my system, but even there I'm getting ATTR_SUPPORTED_FEATURES not in attributes.

```
  File "/home/runner/work/adaptive-lighting/adaptive-lighting/core/homeassistant/components/adaptive_lighting/switch.py", line 560, in _supported_features
    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
KeyError: 'supported_features'
```
see https://github.com/basnijholt/adaptive-lighting/actions/runs/4528230278/jobs/7974822006